pkg/ipam/allocator.go

```diff
@@ -171,32 +193,36 @@ func (ipam *IPAM) ReleaseIPString(ipAddr string) error {
 }
 
 // Dump dumps the list of allocated IP addresses
-func (ipam *IPAM) Dump() ([]string, []string) {
+func (ipam *IPAM) Dump() (map[string]string, map[string]string) {
 	ipam.allocatorMutex.RLock()
 	defer ipam.allocatorMutex.RUnlock()
 
-	allocv4 := []string{}
+	allocv4 := map[string]string{}
 	ralv4 := k8sAPI.RangeAllocation{}
 	if ipam.IPv4Allocator != nil {
 		ipam.IPv4Allocator.Snapshot(&ralv4)
 		origIP := big.NewInt(0).SetBytes(ipam.nodeAddressing.IPv4().AllocationCIDR().IP.To4())
 		v4Bits := big.NewInt(0).SetBytes(ralv4.Data)
 		for i := 0; i < v4Bits.BitLen(); i++ {
 			if v4Bits.Bit(i) != 0 {
-				allocv4 = append(allocv4, net.IP(big.NewInt(0).Add(origIP, big.NewInt(int64(uint(i+1)))).Bytes()).String())
+				ip := net.IP(big.NewInt(0).Add(origIP, big.NewInt(int64(uint(i+1)))).Bytes()).String()
+				owner, _ := ipam.owner[ip]
```

Comment:
Did you intend to check existence here? (just asking, it was a bit surprising to see the `, _` bit)

Reply:
If the owner is not available, the IP is still reported but with an empty/nil owner. Adding a comment.